applications/audio/broadcast_configuration_tool/src/main.c

```diff
@@ -378,105 +440,428 @@ static int cmd_list(const struct shell *shell, size_t argc, char **argv)
+}
+
+static int cmd_preset(const struct shell *shell, size_t argc, char **argv)
+{
+	if (argc < 3) {
+		shell_error(shell, "Usage: bct preset <preset> <BIG index>");
+		return -EINVAL;
+	}
+
+	struct bt_bap_lc3_preset *preset = preset_find(argv[1]);
+
+	if (!preset) {
+		shell_error(shell, "Preset not found");
+		return -EINVAL;
+	}
+
+	size_t big_index = strtoul(argv[2], NULL, 10);
+
+	if (big_index >= CONFIG_BT_ISO_MAX_BIG) {
+		shell_error(shell, "BIG index out of range");
+		return -EINVAL;
+	}
+
+	broadcast_create(big_index);
+
+	if (argc == 4) {
+		size_t sub_index = strtoul(argv[3], NULL, 10);
+
+		if (sub_index >= CONFIG_BT_BAP_BROADCAST_SRC_SUBGROUP_COUNT ||
+		    (sub_index >= broadcast_param[big_index].num_subgroups)) {
+			shell_error(shell, "Subgroup index out of range");
+			return -EINVAL;
 		}
 
-		ret = bt_mgmt_per_adv_stop(i);
-		if (ret) {
-			shell_error(shell, "Failed to stop periodic advertiser");
-			return ret;
+		broadcast_param[big_index].subgroups[sub_index].group_lc3_preset = *preset;
+		broadcast_param[big_index].subgroups[sub_index].preset_name =
+			preset_name_find(preset);
+	} else {
+		for (size_t i = 0; i < broadcast_param[big_index].num_subgroups; i++) {
+			broadcast_param[big_index].subgroups[i].group_lc3_preset = *preset;
+			broadcast_param[big_index].subgroups[i].preset_name =
+				preset_name_find(preset);
 		}
+	}
 
-		ret = bt_mgmt_ext_adv_stop(i);
-		if (ret) {
-			shell_error(shell, "Failed to stop extended advertiser");
-			return ret;
+	return 0;
+}
+
+static int cmd_packing(const struct shell *shell, size_t argc, char **argv)
+{
+	if (argc < 2) {
+		shell_error(shell, "Usage: bct packing <seq/int> <BIG index>");
+		return -EINVAL;
+	}
+
+	size_t big_index = strtoul(argv[1], NULL, 10);
+
+	if (big_index >= CONFIG_BT_ISO_MAX_BIG) {
+		shell_error(shell, "BIG index out of range");
+		return -EINVAL;
+	}
+
+	if (strcmp(argv[1], "seq") == 0) {
+		broadcast_param[big_index].packing = BT_ISO_PACKING_SEQUENTIAL;
+	} else if (strcmp(argv[1], "int") == 0) {
+		broadcast_param[big_index].packing = BT_ISO_PACKING_INTERLEAVED;
+	} else {
+		shell_error(shell, "Invalid packing type");
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int cmd_lang_set(const struct shell *shell, size_t argc, char **argv)
+{
+	if (argc < 4) {
+		shell_error(shell, "Usage: bct lang_set <language> <BIG index> <subgroup index>");
+		return -EINVAL;
+	}
+
+	char language[3];
+
+	memcpy(language, argv[1], 3);
+
+	size_t big_index = strtoul(argv[2], NULL, 10);
+
+	if (big_index >= CONFIG_BT_ISO_MAX_BIG) {
+		shell_error(shell, "BIG index out of range");
+		return -EINVAL;
+	}
+
+	size_t sub_index = strtoul(argv[3], NULL, 10);
+
+	if (sub_index >= CONFIG_BT_BAP_BROADCAST_SRC_SUBGROUP_COUNT) {
+		shell_error(shell, "Subgroup index out of range");
+		return -EINVAL;
+	}
+
+	bt_audio_codec_cfg_meta_set_stream_lang(
+		&broadcast_param[big_index].subgroups[sub_index].group_lc3_preset.codec_cfg,
+		(uint32_t)sys_get_le24(language));
```

Comment:
```suggestion
		sys_get_le24(language));
```
`sys_get_le24` seems to an `uint32_t` - did you get errors without the cast?

Reply:
Will test it out